### ISSUE TITLE
`Attempt to invoke virtual method 'long com.owncloud.android.datamodel.OCFile.getParentId()' on a null object`

### ISSUE THREAD

**Issue by FungY911** [nextcloud/android#13060]
### Cause of error
```java
Exception in thread "Thread-39" java.lang.NullPointerException: Attempt to invoke virtual method 'long com.owncloud.android.datamodel.OCFile.getParentId()' on a null object reference
    at com.owncloud.android.operations.RefreshFolderOperation.synchronizeData(RefreshFolderOperation.java:458)
    at com.owncloud.android.operations.RefreshFolderOperation.fetchAndSyncRemoteFolder(RefreshFolderOperation.java:418)
    at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:247)
    at com.owncloud.android.lib.common.operations.RemoteOperation.run(RemoteOperation.java:387)
    at java.lang.Thread.run(Thread.java:1012)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30290090`
* Build flavor: `gplay`

### Device information
* Brand: `samsung`
* Device: `a52q`
* Model: `SM-A525F`
* Id: `UP1A.231005.007`
* Product: `a52qnseea`

### Firmware
* SDK: `34`
* Release: `14`
* Incremental: `A525FXXS6EXC9`

**Comment by joshtrichards** [comment]
Reproduction steps?

**Comment by joshtrichards** [comment]
Fixed in #13084
